opencog/rule-engine/backwardchainer/BackwardChainer.cc

```diff
@@ -197,7 +197,17 @@ void BackwardChainer::fulfill_fcs(const Handle& fcs)
 	// results
 	AtomSpace tmp_as(&_as);
 
-	// Run the FCS and add the results in _as
+	// Run the FCS and add the results, if any, in _as.
+	//
+	// Warning: since tmp_as is a child of _as, TVs of existing atoms
+	// in _as, that are modified by running fcs will be modified on
+	// _as as well. This can create involontary TVs changes, hopefully
+	// mitigated by the merging the TVs properly (for now the one with
+	// the highest confidence wins). To avoid that side effect, we
+	// could operate on a copy the atomspace, of its zone of focus. Or
+	// alternatively modify some HypotheticalLink wrapping the atoms
+	// of concerns instead of the atoms themselves, and only modify
+	// the atoms if there are existing results to copy back to _as.
 	Handle hresult = bindlink(&tmp_as, fcs);
```

Comment:
I think to do such reasoning we would wrap the "Assume X" in a context, implication or inheritance link, to avoid being temporarily inconsistent with the atomspace. So the context links or such would be used to essentially describe kripke frames.